src/pages/Finances.tsx

```diff
@@ -8,6 +11,26 @@ import FinancialOverview from '../features/financial-overview/FinancialOverview'
 import { MainGridWrapper, MainGrid, MainGridCol, SecondGridWrapper } from '../ui/PageGrid';
 
 const Finances = () => {
+
+  const announcementType = 'financial'
+  const [events, setEvents] = useState<any>([]);
+  const user = useContext<any>(UserContext);
+  const announcements = useAnnouncements(announcementType);
+  
+  /* eslint-disable react-hooks/exhaustive-deps */
+  // Fetch data on load
+  useEffect(() => {
+    let announcementsToUse: any[] = [];
+
+    if (!user.loading && !announcements.loading) {
+      announcementsToUse = announcements.data.filter(announcement =>
+        hasAudience(user.data, announcement)
+      );
+    }
+    setEvents(announcementsToUse);
```

Comment:
Because `events` state is defaulting to an empty array, you don't need the concept of a mutable array here.. just check if the user is done loading, the announcements are done loading and then `setEvents(announcements.data.filter......)` .. just keeps it simple.